### ISSUE TITLE
Defect: Resolving library path variable in the generated script setup.sh

### ISSUE THREAD

**Issue by dtabell** [sourceryinstitute/OpenCoarrays#629]
[issue stat img]: https://img.shields.io/issuestats/i/github/sourceryinstitute/OpenCoarrays.svg?style=flat-square

| Avg response time              |
|:------------------------------:|
| ![Issue Stats][issue stat img] |

# Defect/Bug Report #

- OpenCoarrays Version: caf version 2.4.0-6-gebc6f30
- Fortran Compiler: gfortran 8.2.0
- C compiler used for building lib: gcc 8.2.0
- Installation method: `install.sh`
- Output of `uname -a`: Linux v.radia.run 4.18.19-100.fc27.x86_64 #1 SMP Wed Nov 14 22:04:34 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- MPI library being used: MPICH 3.3
- Machine architecture and number of physical cores: MacBookPro running a Linux VM
- Version of CMake: cmake version 3.11.2

## Observed Behavior ##
In my OpenCoarrays setup.sh—which is generated automatically during the build and install process—I see the following stanza:
```
   # Prepend the compiler library paths to the LD_LIBRARY_PATH environment variable:
   if [[ -z "${!LD_LIB_P_VAR}" ]]; then
     export LD_LIBRARY_PATH="/opt/gcc/8.2.0/lib64/:/opt/gcc/8.2.0/lib"
   else
     export
   LD_LIBRARY_PATH="/opt/gcc/8.2.0/lib64/:/opt/gcc/8.2.0/lib:${!LD_LIB_P_VAR}"
   fi
```
## Expected Behavior ##
The `${!LD_LIB_P_VAR}` should read `$LD_LIBRARY_PATH`.

On my machine (a linux VM) `${!LD_LIB_P_VAR}` does not resolve to anything, which means the if test takes the first branch and overwrites the previous contents of `$LD_LIBRARY_PATH`. I suspect you need `${!LD_LIB_P_VAR}` and its associated logic higher up the chain, but that variable should be resolved to the appropriate value before its written to the _generated_ setup.sh file.

While you're at it, you might as well also delete the trailing `/` after `lib64`, so that, for example, the penultimate line reads
```
  LD_LIBRARY_PATH="/opt/gcc/8.2.0/lib64:/opt/gcc/8.2.0/lib:$LD_LIBRARY_PATH"
```

[links]:#
[Google]: https://www.google.com
[Stack Overflow]: https://stackoverflow.com
[fenced codeblocks]: https://help.github.com/articles/creating-and-highlighting-code-blocks/
[GFortran bug tracker]: https://gcc.gnu.org/bugzilla/
[Intel MPI]: https://software.intel.com/en-us/intel-mpi-library
[MPT]: http://www.sgi.com/products/software/sps.html
[MVAPICH]: http://mvapich.cse.ohio-state.edu
[MPICH]: https://www.mpich.org
[Open-MPI]: https://www.open-mpi.org
[CONTRIBUTING.md]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/CONTRIBUTING.md
[Helping Out]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/CONTRIBUTING.md#helping-out
[official release]: https://github.com/sourceryinstitute/OpenCoarrays/releases
[CMake install]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/INSTALL.md#cmake-scripts
[homebrew]: https://brew.sh
[issues page]: https://github.com/sourceryinstitute/OpenCoarrays/issues
[mailing list]: https://groups.google.com/forum/#!forum/opencoarrays
[install.sh]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/install.sh

**Comment by zbeekman** [comment]
@rouson did the fix not work? Is that why you re-opened it?

**Comment by rouson** [comment]
@zbeekman The previous fix left extraneous `!` characters in `setup.sh` and those caused `source setup.sh` to return an error.  Deleting the `!` characters generates a `setup.sh` that works for me.  I've done this in PR #641.

**Comment by zbeekman** [comment]
@rouson yeah, I just saw your PR. doh! I thought my logic was correct, but you are definitely right; the `!` needs to go.

**Comment by zbeekman** [comment]
we can create a patch release while you're here.
